train_segmentation_model.py

```diff
@@ -89,17 +89,50 @@ def train(gcp_bucket, config_file, random_module_global_seed, numpy_random_globa
     validation_generator = ImagesAndMasksGenerator(
         Path(local_dataset_dir, train_config['dataset_id'],
              'validation').as_posix(),
-        rescale=1./255,
+        rescale=1. / 255,
         target_size=target_size,
         batch_size=batch_size,
         seed=None if 'validation_data_shuffle_seed' not in train_config else train_config['validation_data_shuffle_seed'])
 
+    if pretrained_model_id is not None:
+        local_pretrained_model_dir = Path(tmp_directory, 'pretrained_models')
+        copy_folder_locally_if_missing(os.path.join(gcp_bucket, 'models', pretrained_model_id), local_pretrained_model_dir)
+        path_pretrained_model = Path(local_pretrained_model_dir, pretrained_model_id, "model.hdf5").as_posix()
+
+        with Path(local_pretrained_model_dir, pretrained_model_id, 'config.yaml').open('r') as f:
+            pretrained_model_config = yaml.safe_load(f)['train_config']
+
+        with Path(local_pretrained_model_dir, pretrained_model_id, 'metadata.yaml').open('r') as f:
+            pretrained_model_metadata = yaml.safe_load(f)
+
+        pretrained_info = {'pretrained_model_id': pretrained_model_id, 'pretrained_config': pretrained_model_config, 'pretrained_metadata': pretrained_model_metadata}
+
+        # confirm that the same model and backbone are used from pretrain and train
+        print(pretrained_model_config['segmentation_model']['model_name'], train_config['segmentation_model']['model_name'])
+        assert pretrained_model_config['segmentation_model']['model_name'] == train_config['segmentation_model']['model_name']
+        print(pretrained_model_config['segmentation_model']['model_parameters']['backbone_name'], train_config['segmentation_model']['model_parameters']['backbone_name'])
+        assert pretrained_model_config['segmentation_model']['model_parameters']['backbone_name'] == train_config['segmentation_model']['model_parameters']['backbone_name']
+        # same loss function
+        print(pretrained_model_config['loss'], train_config['loss'])
+        assert pretrained_model_config['loss'] == train_config['loss']
+        # confirm that the number of classes in pretrain is the same as train
+        print(pretrained_model_metadata['num_classes'], len(train_generator.mask_filenames))
+        assert pretrained_model_metadata['num_classes'] == len(train_generator.mask_filenames)
+        # same target size
+        print(pretrained_model_metadata['dataset_config']['target_size'], dataset_config['target_size'])
+        assert pretrained_model_metadata['dataset_config']['target_size'] == dataset_config['target_size']
```

Comment:
last thing i can think of is to assert number of color channels (rgb vs. gv). somewhere in the code there is a hint how to check this

Reply:
first layer of model would tell this, but accessing these layer dimensions may not be straightforward, so can skip for now. a model on gv will break with rgb. currently it's hardcoded for only GV (`model = Unet(input_shape=(None, None, 1),` in `models.py`), so we'll revisit this assertion (hopefully we remember) if/when we open up the code to accept rgb (via train_config)